flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/aggfunctions/FirstValueAggFunction.java

```diff
@@ -83,7 +83,7 @@ public void accumulate(RowData rowData, Object value) {
 		GenericRowData acc = (GenericRowData) rowData;
 		if (value != null && acc.getLong(1) == Long.MAX_VALUE) {
 			acc.setField(0, value);
-			acc.setField(1, System.currentTimeMillis());
+			acc.setField(1, System.nanoTime());
```

Comment:
I prefer the higher precision of nanoTime, for we use this as the order when merging.

Reply:
The `System.nanoTime()` has higher precision but it can not be used in different JVMs because every JVM because there's no guarantee that nanoTime is relative to any timebase.